snaps_boot/provision/hardware/pxe_utils.py

```diff
@@ -44,15 +44,67 @@ def __main(config, operation):
     bmc_dict = prov_dict.get('BMC')
     subnet_list = dhcp_dict.get('subnet')
     cpu_core_dict = prov_dict.get('CPUCORE')
-
+    pxe_server_configuration_listmap = tftp_dict.get('pxe_server_configuration')
+    buildPxeServer = None
+    ubuntuPxeServer = False
+    centosPxeServer = False
+    ubuntu_dict = {}
+    if pxe_server_configuration_listmap == None:
+       logger.error("Please define OS(ubuntu/centos) details in hosts.yaml to create pxe server")
+       exit(1)
+    for item in pxe_server_configuration_listmap:
+      for key in item:
+        if ("ubuntu" == key):
+           ubuntuPxeServer = True
+           buildPxeServer = "ubuntu"
+           ubuntu_dict = item.get("ubuntu")
+        if ("centos" == key):
+           centosPxeServer = True
+           buildPxeServer = "centos"
+           centos_dict = item.get("centos")
+
+    if(True == ubuntuPxeServer and True == centosPxeServer):
+      buildPxeServer = "ubuntu + centos"
+
+    logger.info("buildPxeServer is :"+str(buildPxeServer))
```

Comment:
I'm finally getting into the swing of string formatting in Python, but the preferred approach is to use "{}{}".format(str1, str2) and we should be moving our code into that direction